### PR TITLE
Upgrade compaction

### PR DESCRIPTION
**Pull request by jmckulk**
This change updates the crunchy-upgrade container to be based off of
crunchy-base image instead of crunchy-pg-base. This is one step closer to
removing the crunchy-pg-base image entirely. The binary and config paths are
also updated to use `/opt/crunchy` instead of `/opt/cpm`.

This set of changes also includes two bug fixes for the upgrade container:

First, the pgnodemx package has been added to the crunchy-postgres image and is now
included in the default postgresql.conf as an included shared object.
pg_upgrade needs access to any custom shared object files that are installed in
the "old" database. The upgrade image now includes the pgnodemx package for all
versions of postgres that are supported by crunchy repos.

There was also a permissions issue that caused the upgrade job to fail. pg_upgrade 
expects the old pgdata directory to have u=rwx (0700) permissions.
This change updates the permissions in the upgrade/start.sh to meet this
expectation and complete the upgrade.
[ch4854]

**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?

Tested centos7 and ubi8 in gke, tested ubi8 in gcp (ocp 3.11). Was able to follow upgrade example docs to complete a major upgrade (with test data)

**Type of Changes:**

 
 - [x] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**

**What is the new behavior (if this is a feature change)?**

**Other information**:
[ch9428]

### CODE REVIEW

File: bin/upgrade/start.sh [hunk 1]
```diff
@@ -97,7 +98,7 @@ case $OLD_VERSION in
     ;;
 esac
 
-export PATH=/opt/cpm/bin:${PGBINNEW?}:$PATH
+export PATH=/opt/crunchy/bin:${PGBINNEW?}:$PATH
```
Comment (jmckulk) [inline bin/upgrade/start.sh]: needs to user CRUNCHY_DIR

Comment (jkatz) [reply bin/upgrade/start.sh]: ^

File: bin/upgrade/start.sh [hunk 2]
```diff
@@ -132,6 +133,7 @@ rm ${PGDATAOLD?}/postmaster.pid
 # changing to /tmp is necessary since pg_upgrade has to have write access
 cd /tmp
 
+chmod 0700 ${PGDATAOLD?}
```
Comment (jkatz) [inline bin/upgrade/start.sh:137]: Should leave a comment here as this is an important line.

**Review by jkatz — CHANGES_REQUESTED** [review]

File: bin/upgrade/start.sh [hunk 3]
```diff
@@ -97,7 +98,7 @@ case $OLD_VERSION in
     ;;
 esac
 
-export PATH=/opt/cpm/bin:${PGBINNEW?}:$PATH
+export "PATH=${CRUNCHY_DIR}/bin:${PGBINNEW?}:$PATH"
```
Comment (jkatz) [inline bin/upgrade/start.sh]:
```suggestion
export PATH="${CRUNCHY_DIR}/bin:${PGBINNEW?}:${PATH}"
```

**Review by jkatz — CHANGES_REQUESTED** [review]

**Review by jkatz — APPROVED** [review]

**Pull request merged**